### PR TITLE
`stop_warp` & `stop_roll` cheatcodes

### LINKED ISSUE #1335: Implement `stop_warp` and `stop_roll`

### PR DESCRIPTION
**Pull request by Arcticae**
Closes #1335

### CODE REVIEW

**Comment by karol-bisztyga** [comment]
This looks ok but I'm wondering:
- what's wrong with the idea of returning the stop callback from the original cheat code?
- are we going to split the logic for other cheat codes like expect call/event as well?

**Comment by Arcticae** [comment]
> * what's wrong with the idea of returning the stop callback from the original cheat code?

it's not going to be doable in cairo 1.0, since there are no hints

> * are we going to split the logic for other cheat codes like expect call/event as well?

yes, that is the idea. i mentioned somewhere a logic like `setTimeout`/`clearTimeout` - should be pretty similar (returning an id, and using separate function to cancel)

File: protostar/cheatable_starknet/callable_hint_locals/roll_hint_local.py [hunk 1]
```diff
@@ -16,10 +16,10 @@ def _build(self) -> Callable[..., Any]:
 
     def roll(
         self,
-        target_contract_address: RawAddress,
+        target_contract_address: int,
         blk_number: int,
-    ) -> Callable[[], None]:
-        return self.controllers.block_info.cheat(
+    ):
```
Comment (karol-bisztyga) [inline protostar/cheatable_starknet/callable_hint_locals/roll_hint_local.py:21]:
```suggestion
    ) -> None:
```
nit, maybe it's not the most important thing in the universe but removing the return type just feels like a step back.

File: protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py [hunk 2]
```diff
@@ -16,10 +14,10 @@ def _build(self):
 
     def warp(
         self,
-        target_contract_address: RawAddress,
+        target_contract_address: int,
         blk_timestamp: int,
-    ) -> Callable[[], None]:
-        return self.controllers.block_info.cheat(
+    ):
+        self.controllers.block_info.cheat(
```
Comment (karol-bisztyga) [inline protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py:20]: Why is this method called `cheat`? Couldn't we use something like `set_block_timestamp` + `set_block_number` instead?

File: protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py [hunk 3]
```diff
@@ -16,10 +14,10 @@ def _build(self):
 
     def warp(
         self,
-        target_contract_address: RawAddress,
+        target_contract_address: int,
         blk_timestamp: int,
-    ) -> Callable[[], None]:
-        return self.controllers.block_info.cheat(
+    ):
```
Comment (karol-bisztyga) [inline protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py:19]:
```suggestion
    ) -> None:
```
nit, maybe it's not the most important thing in the universe but removing the return type just feels like a step back.

**Review by karol-bisztyga — APPROVED** [review]
Ok, cool, thanks for the answer, this code LGTM mostly, maybe it'd be good to get one more review for this.

File: protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py (hunk 2 above)
Comment (Arcticae) [reply protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py:20]: It's a legacy thing, it was copied over. If you feel like it, we can change the structure in this PR.

Comment (Arcticae) [reply protostar/cheatable_starknet/callable_hint_locals/warp_hint_local.py]: https://github.com/software-mansion/protostar/pull/1429/commits/3f458f11e2f42aa50d0590e2b5ecb963d10185e1

**Pull request merged**